### ISSUE TITLE
Compile: Support generating PDFs

### ISSUE THREAD

**Issue by kevboh** [kevboh/longform#1]
Compile should be able to turn a draft into a single PDF file.

**Comment by Edo78** [comment]
I don't think that this plugin should be too bloated. There are always plugin for such feature (like the pandoc one).
IMO you should think only about the features unique of this plugin

**Comment by kevboh** [comment]
If anyone is interested, this can be done right now (and has been doable for a while) via custom compile steps and delegating to something like Pandoc. I'm still interested in adding a dedicated "convert to PDF" compile step, however, although it will never be as full-featured as Pandoc.

**Comment by nomaad** [comment]
What about a generic terminal-step that calls a terminal command and receives the path to the final manuscript as an argument? Like this Pandoc could be called with any output format desired.

**Comment by nonsciencemark** [comment]
> If anyone is interested, this can be done right now (and has been doable for a while) via custom compile steps and delegating to something like Pandoc. I'm still interested in adding a dedicated "convert to PDF" compile step, however, although it will never be as full-featured as Pandoc.

Hi kevboh, I'm late to the party here, but do you have an example on hand of how we can pass to Pandoc during compilation? FYI I mainly need to detect and format citations using Pandoc and then include them as a bibliography at the end, but I'm not at all familiar with Javascript so I have no idea where to start

**Comment by kevboh** [comment]
> > If anyone is interested, this can be done right now (and has been doable for a while) via custom compile steps and delegating to something like Pandoc. I'm still interested in adding a dedicated "convert to PDF" compile step, however, although it will never be as full-featured as Pandoc.
> 
> 
> 
> Hi kevboh, I'm late to the party here, but do you have an example on hand of how we can pass to Pandoc during compilation? FYI I mainly need to detect and format citations using Pandoc and then include them as a bibliography at the end, but I'm not at all familiar with Javascript so I have no idea where to start

I don't have anything so complex, unfortunately. I'm not super familiar with pandoc's options, but yes, wiring it together will require some JS knowledge at present. Sorry I can't be of more help--just too busy at present to do feature work.

**Comment by nonsciencemark** [comment]
> > > If anyone is interested, this can be done right now (and has been doable for a while) via custom compile steps and delegating to something like Pandoc. I'm still interested in adding a dedicated "convert to PDF" compile step, however, although it will never be as full-featured as Pandoc.
> > 
> > 
> > Hi kevboh, I'm late to the party here, but do you have an example on hand of how we can pass to Pandoc during compilation? FYI I mainly need to detect and format citations using Pandoc and then include them as a bibliography at the end, but I'm not at all familiar with Javascript so I have no idea where to start
> 
> I don't have anything so complex, unfortunately. I'm not super familiar with pandoc's options, but yes, wiring it together will require some JS knowledge at present. Sorry I can't be of more help--just too busy at present to do feature work.

No problem! Thanks for the cool plugin
